### PR TITLE
Fix RealmQuery scope issue

### PR DESCRIPTION
**Pull request by cmelchior**
This PR https://github.com/realm/realm-java/pull/721 seems to have introduced a regression causing subqueries on the RealmResult to use the entire table instead of the results of the previous query.

### CODE REVIEW

**Comment by cmelchior** [comment]
@emanuelez @kneth @bmunkholm

File: realm/src/main/java/io/realm/RealmQuery.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
 public class RealmQuery<E extends RealmObject> {
 
     private Realm realm;
+    private Table tableSpec;
```
Comment (kneth) [inline realm/src/main/java/io/realm/RealmQuery.java]: Please don't name it `tableSpec` as we have an internal class called `TableSpec`.

**Comment by kneth** [comment]
Please update the change log.

**Comment by cmelchior** [comment]
There shouldn't be any need. It is a regression that hasn't been in a public release yet. There is a changelog entry for the original error that triggered this ;)

File: realm/src/main/java/io/realm/RealmQuery.java (hunk 1 above)
Comment (emanuelez) [inline realm/src/main/java/io/realm/RealmQuery.java]: I agree. `table` would be a better name IMO

File: realm/src/main/java/io/realm/RealmQuery.java [hunk 2]
```diff
@@ -79,14 +78,10 @@ public RealmQuery(RealmResults realmList, Class<E> clazz) {
...
-        TableOrView dataStore = getTable();
+        this.tableSpec = realm.getTable(clazz);
         this.columns = Realm.columnIndices.get(clazz.getSimpleName());
     }
 
-    TableOrView getTable() {
-        return realm.getTable(clazz);
-    }
-
     private boolean containsDot(String s) {
```
Comment (emanuelez) [inline realm/src/main/java/io/realm/RealmQuery.java]: Not related to the PR but I wonder why this is here. And why not using s.contains('.'). The compiler should do all the needed optimizations

Comment (cmelchior) [inline realm/src/main/java/io/realm/RealmQuery.java]: Yeah I stumble across it a few times now, and forgot to refactor it each time.

**Comment by emanuelez** [comment]
:+1: apart from the comment. No need for line in changelog IMO

File: realm/src/main/java/io/realm/RealmQuery.java (hunk 2 above)
Comment (bmunkholm) [inline realm/src/main/java/io/realm/RealmQuery.java]: @cmelchior Then fix it now.

**Comment by bmunkholm** [comment]
Otherwise :+1:

**Pull request merged**